src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1617,6 +1673,32 @@ public override IOperation VisitCatchClause(ICatchClauseOperation operation, int
             throw ExceptionUtilities.Unreachable;
         }
 
+        public override IOperation VisitReturn(IReturnOperation operation, int? captureIdForResult)
+        {
+            Debug.Assert(_currentStatement == operation);
+            IOperation returnedValue = Visit(operation.ReturnedValue);
+
+            switch (operation.Kind)
+            {
+                case OperationKind.YieldReturn:
+                    AddStatement(new ReturnStatement(OperationKind.YieldReturn, returnedValue, semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit));
```

Comment:
So just to confirm, we will still retain IReturnOperation in the CFG for YieldReturn cases OR do we plan to change this in future?

Reply:
> So just to confirm, we will still retain IReturnOperation in the CFG for YieldReturn cases OR do we plan to change this in future?

We will retain those because they do not change logical control flow within the function, the next statement is getting executed after it. 

---
In reply to: [171986854](https://github.com/dotnet/roslyn/pull/25190#discussion_r171986854) [](ancestors = 171986854)